src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                return new Square(color, random.nextDouble() * 10);
+            case 1:
+                return new Rectangle(color, random.nextDouble() * 10, random.nextDouble() * 10);
+            case 2:
+                return new RightTriangle(color, random.nextDouble() * 10, random.nextDouble() * 10);
+            case 3:
+                return new Circle(color, random.nextDouble() * 10);
+            case 4:
+                return new IsoscelesTrapezoid(color, random.nextDouble()
+                        * 10, random.nextDouble() * 10, random.nextDouble() * 10);
```

Comment:
Avoid passing expressions directly in the constructor. Instead, create separate variables for the dimensions and pass those variables to the constructor for better readability.